fiksi/src/solve/lm.rs

```diff
@@ -43,8 +43,8 @@ pub(crate) fn levenberg_marquardt(variables: &mut [f64], subsystem: &Subsystem<'
 
         if residual < prev_residual {
             lambda *= 0.5;
-            if lambda < 1e-10 {
-                lambda = 1e-10;
+            if lambda < 1e-50 {
+                lambda = 1e-50;
             }
         } else {
```

Comment:
I want to implement smarter step acceptance and lambda updating, as the current state isn't optimal: steps in the wrong direction are accepted with lambda being updated after-the-fact, leading to worse convergence performance.